lib/chef/provider/user/dscl.rb

```diff
@@ -329,7 +329,7 @@ def ditto_home
 
         def move_home
           Chef::Log.debug("#{new_resource} moving #{self} home from #{current_resource.home} to #{new_resource.home}")
-
+          new_resource.gid(20) if new_resource.gid.nil?
```

Comment:
For a Mac system imho 20 is a reasonable default, because if anyone is creating a "special" system user or admin user they would most likely be explicit and set the gid.

This is certainly an edge case but my point is, the user resource allows anyone to create a user with his being required even when a managed home is in use.

You are already setting it to 20 if gid is nil in the dscl_set_gid function.

Chef doesn't document it anywhere I can see.

I think an alternative to my PR would be to raise an exception if gid is not specified, ie. make it a requirement in the user resource.